content/blogposts/2023/copilot-vs-cody.md

```diff
@@ -0,0 +1,229 @@
+
+![Codebase npm scripts](https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/npm-scripts.png)
+
+With Cody, if we followed the instructions, we would have our local environment up and running. In contrast, with Copilot, we would run into issues ranging from missing environment variables and missing database tables to our frontend not building at all. We have to give Cody the win here as well. Our score now stands at:
+
+Cody - 2
+
+Copilot - 0
+
+## Scenario: Familiarize yourself with the code
+
+Now that we know a bit about what the application does and have it running locally, let's ask our code AI assistants some questions about how the actual application functions. In this application, we'll be working a lot with video files, so let's ask about the current capabilities that exist. The first question we'll ask is: **"What are the properties on the video model?"**
+
+![Explain feature of a codebase](https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/explain-feature-of-codebase.png)
+
+Cody finds the `Video` model and provides us with the various properties that are present on it. Copilot states that it does not have direct access to our codebase and therefore does not know what properties exist on the `Video` model. It does try to guess and assume some properties that may or may not exist.
+
+Getting the properties of a model is neat, but not something to necessarily write home about. Let's try a more tricky question. In this project, we spend a lot of time working with LLMs and one of the most important things is keeping track of costs. Let's ask our code AI assistants how we can calculate our costs for a video. The query will be: **"How do I calculate the OpenAI cost for a video?"**
+
+![Explain calculating costs of OpenAI call](https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/calculate-costs-explain.png)
+
+Cody is able to find an existing function called `calculateOpenAICost()` and gives us a detailed explanation of what this function does, how it works, and where it's located if we want to learn more. Copilot on the other hand gives us a general answer on how to calculate the cost of an OpenAI call and some pricing information. With these two examples, I think we can give Cody the win here as well.
+
+Cody - 3
+
+Copilot - 0
+
+## Scenario: Adding new functionality to our codebase
+
+Now we get to the fun parts - writing code. As developers, we are accustomed to having our IDE making it easier to write code through syntax highlighting, intellisense, and other enhancements that make it easier for us to focus on the hard parts. Code AI assistants can go a step further here in writing some or a lot of the code for us.
```

Comment:
"IntelliSense" is stylized in VS Code's docs - should it be here, or is it lowercase when used more generally?

```suggestion
Now we get to the fun parts - writing code. As developers, we are accustomed to having our IDE making it easier to write code through syntax highlighting, IntelliSense, and other enhancements that make it easier for us to focus on the hard parts. Code AI assistants can go a step further here in writing some or a lot of the code for us.
```